tests/python_functional/functional_tests/parsers/regexp-parser/test_regexp_parser.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+#############################################################################
+# Copyright (c) 2020 One Identity
+# Copyright (c) 2021 Xiaoyu Qiu
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+import pytest
+
+
+test_parameters_raw = [
+    ("foo", '''""''', '''"(?<key>foo)"''', "", "${key}", True, True, "foo"),
+    ("foo", '''""''', '''"(?<key>fo*)"''', "", "${key}", True, True, "foo"),
+    ("foo", '''"reg"''', '''"fo*"''', "", "${reg}", True, True, "foo"),
+    ("foo", '''".reg."''', '''"(?<key>foo)"''', "", "${.reg.key}", True, True, "foo"),
+    ("foo", '''".reg."''', '''"(?<key>foo)"''', "", "${key}", True, True, ""),
+    ("foo", '''".reg."''', '''"(?<key>foo)|(?<key>bar)"''', "dupnames", "${.reg.key}", True, True, "foo"),
+    ("abc", '''""''', '''"(?<key>Abc)"''', "", "${key}", True, False, ""),
+    ("abc", '''""''', '''"(?<key>Abc)"''', "ignore-case", "${key}", True, True, "abc"),
+    ("foobar", '''".reg."''', '''"(?<key>foo)", "(?<key>bar)"''', "", "${.reg.key}", True, True, "foo"),
+    ("foo", '''""''', '''"(?<key>foo"''', "", "${key}", False, False, ""),
+    ("foo", '''""''', '''"fo*"''', "", "$MSG", True, True, "foo"),
+]
+
+
+@pytest.mark.parametrize(
+    "input_message, prefix, patterns, flags, template, compile_result, expected_result, expected_value", test_parameters_raw,
+    ids=[
+        "match_literally", "match_regular_expression", "match_without_reference_saving",
+        "match_with_prefix", "match_with_prefix_and_wrong_template", "match_with_dupnames", "unmatch_case",
+        "match_ignore_case_flag", "match_multiple_regular_expressions", "regular_expression_compile_error",
+        "check_MSG_macro",
+    ],
+)
+def test_regexp_parser(config, syslog_ng, syslog_ng_ctl, input_message, prefix, patterns, flags, template, compile_result, expected_result, expected_value):
```

Comment:
Thanks for the changes, they are very good.
My comment here: syslog_ng_ctl fixture is not used in the testcase, it can be deleted from testcase parameter list